cmd/hamctl/command/promote.go

```diff
@@ -21,43 +18,34 @@ func NewPromote(client *httpinternal.Client, service *string) *cobra.Command {
 			})
 		},
 		RunE: func(c *cobra.Command, args []string) error {
-			committerName, committerEmail, err := git.CommitterDetails()
-			if err != nil {
-				return err
+			if fromEnvironment == "" {
+				switch toEnvironment {
+				case "dev":
+					fromEnvironment = "master"
+				case "staging":
+					fromEnvironment = "dev"
+				case "prod":
+					fromEnvironment = "staging"
+				}
 			}
-
-			var resp httpinternal.PromoteResponse
-			url, err := client.URL("promote")
-			if err != nil {
-				return err
+			var artifactID string
+			var err error
+			if fromEnvironment == "master" {
+				artifactID, err = actions.ArtifactIDFromBranch(client, *service, "master")
+			} else {
+				artifactID, err = actions.ArtifactIDFromEnvironment(client, *service, namespace, fromEnvironment)
 			}
-			err = client.Do(http.MethodPost, url, httpinternal.PromoteRequest{
-				Service:        *service,
-				Environment:    environment,
-				Namespace:      namespace,
-				CommitterName:  committerName,
-				CommitterEmail: committerEmail,
-			}, &resp)
 			if err != nil {
 				return err
 			}
-			fmt.Printf("Promote service: %s\n", *service)
-			if resp.Status != "" {
-				fmt.Printf("%s\n", resp.Status)
-			} else {
-				fmt.Printf("[✓] Promotion of %s from %s to %s initialized\n", resp.Tag, resp.FromEnvironment, resp.ToEnvironment)
-			}
 
-			return nil
+			return actions.ReleaseArtifactID(client, *service, toEnvironment, artifactID, intent.NewPromoteEnvironment(fromEnvironment))
 		},
 	}
-	command.Flags().StringVarP(&environment, "env", "e", "", "Environment to promote to (required)")
-	completion.FlagAnnotation(command, "env", "__hamctl_get_environments")
-	// errors are skipped here as the only case they can occour are if thee flag
-	// does not exist on the command.
-	//nolint:errcheck
+	command.Flags().StringVarP(&toEnvironment, "env", "", "", "Environment to promote to (required)")
```

Comment:
The alias `e` is removed here. Is that intentional?
```suggestion
	command.Flags().StringVarP(&toEnvironment, "env", "e", "", "Environment to promote to (required)")
```

Reply:
No, lets keep it 👍